packages/ui/src/components/va-data-table/hooks/useTableScroll.ts

```diff
@@ -51,7 +51,9 @@ export default function useTableScroll (
     rootMargin: `${props.scrollTopMargin ?? 0}px 0px ${props.scrollBottomMargin ?? 0}px 0px`,
   }))
 
-  useIntersectionObserver(intersectionHandler, options, targets)
+  onMounted(() => {
+    useIntersectionObserver(intersectionHandler, options, targets)
```

Comment:
This is wrong. All composables must be used in scope of setup function, not mounted.

You will cause making things on each mount, which potentially can cause memory leaks. Especially with intersection observer, which must call disconnect method like destruxtor